crates/ledger/src/store.rs

```diff
@@ -64,50 +108,41 @@ pub trait Store: Send + Sync {
         &mut self,
         epoch: Epoch,
         rewards_summary: Option<RewardsSummary>,
-    ) -> Result<(), Self::Error>;
-
-    /// Get details about a specific Pool
-    fn pool(&self, pool: &PoolId) -> Result<Option<pools::Row>, Self::Error>;
-
-    /// Get details about a specific UTxO
-    fn resolve_input(
-        &self,
-        input: &TransactionInput,
-    ) -> Result<Option<TransactionOutput>, Self::Error>;
+    ) -> Result<(), StoreError>;
 
     /// Compute stake distribution using database snapshots.
-    fn stake_distribution(&self, epoch: Epoch) -> Result<StakeDistribution, Self::Error>;
+    fn stake_distribution(&self, epoch: Epoch) -> Result<StakeDistribution, StoreError>;
 
     /// Compute rewards using database snapshots and a previously computed stake distribution.
     fn rewards_summary(
         &self,
         stake_distribution: StakeDistribution,
-    ) -> Result<RewardsSummary, Self::Error>;
+    ) -> Result<RewardsSummary, StoreError>;
 
     /// Get current values of the treasury and reserves accounts.
     fn with_pots<A>(
         &self,
         with: impl FnMut(Box<dyn BorrowMut<pots::Row> + '_>) -> A,
-    ) -> Result<A, Self::Error>;
+    ) -> Result<A, StoreError>;
 
     /// Provide an access to iterate over pools, in a way that enforces:
     ///
     /// 1. That mutations will be persisted on-disk
     ///
     /// 2. That all operations are consistent and atomic (the iteration occurs on a snapshot, and
     ///    the mutation apply to the iterated items)
-    fn with_pools(&self, with: impl FnMut(pools::Iter<'_, '_>)) -> Result<(), Self::Error>;
+    fn with_pools(&self, with: impl FnMut(pools::Iter<'_, '_>)) -> Result<(), StoreError>;
 
     /// Provide an access to iterate over accounts, similar to 'with_pools'.
-    fn with_accounts(&self, with: impl FnMut(accounts::Iter<'_, '_>)) -> Result<(), Self::Error>;
+    fn with_accounts(&self, with: impl FnMut(accounts::Iter<'_, '_>)) -> Result<(), StoreError>;
 
     /// Provide an iterator over slot leaders, similar to 'with_pools'. Note that slot leaders are
     /// stored as a bounded FIFO, so it only make sense to use this function at the end of an epoch
     /// (or at the beginning, before any block is applied, depending on your perspective).
-    fn with_block_issuers(&self, with: impl FnMut(slots::Iter<'_, '_>)) -> Result<(), Self::Error>;
+    fn with_block_issuers(&self, with: impl FnMut(slots::Iter<'_, '_>)) -> Result<(), StoreError>;
 
     /// Provide an access to iterate over utxo, similar to 'with_pools'.
-    fn with_utxo(&self, with: impl FnMut(utxo::Iter<'_, '_>)) -> Result<(), Self::Error>;
+    fn with_utxo(&self, with: impl FnMut(utxo::Iter<'_, '_>)) -> Result<(), StoreError>;
```

Comment:
I believe we can likely remove that method entirely; I don't think we ever need to iter over UTxOs other than for rewards calculation 🤔.